extension-style/src/main/java/com/mapbox/maps/extension/style/types/PromoteId.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.maps.extension.style.types
+
+import androidx.annotation.Keep
+import com.mapbox.bindgen.Value
+import com.mapbox.maps.extension.style.sources.generated.GeoJsonSource
+import com.mapbox.maps.extension.style.sources.generated.VectorSource
+
+/**
+ * Holds a property type to promote a specific feature for feature state API.
+ *
+ * @param sourceId source layer id of the feature, either source [GeoJsonSource] or [VectorSource].
+ * @param propertyName feature property name.
+ *
+ * For more information see [The online documentation]https://docs.mapbox.com/mapbox-gl-js/style-spec/types/#promoteId).
+ */
+@Keep
+data class PromoteId @JvmOverloads constructor(
+  val sourceId: String? = null,
+  val propertyName: String
+) {
+  internal fun toValue(): Value {
+    sourceId?.let {
+      return Value(hashMapOf(sourceId to Value(propertyName)))
+    }
+    return Value(propertyName)
+  }
+
+  companion object {
+    /**
+     * Construct a [PromoteId] object from a Property returned from the core.
+     * Can be either of type [String] or [HashMap] of <String,String>
+     */
+    fun fromProperty(propertyName: Any): PromoteId {
+      return when (propertyName) {
+        is String -> {
+          PromoteId(propertyName = propertyName)
+        }
+        is HashMap<*, *> -> {
+          @Suppress("UNCHECKED_CAST")
+          val propertyMap = propertyName as HashMap<String, String>
```

Comment:
what if it's not `HashMap<String, String>`? Will we have runtime crash? Perhaps use try-catch here to produce better exception text

Reply:
I'm not sure how it works, could check like:
```
is HashMap<String, String> -> {
```
instead of 
```
is HashMap<*, *> -> {
```
work here?

Reply:
Using HashMap<String,String> gives compile time error.
`Cannot check for instance of erased type: kotlin.collections.HashMap<String, String> /* = java.util.HashMap<String, String>`

Reply:
I will add more try{} catch{} for hashmap, but the args for this function returned from core in either string or map<string,string>.